src/hooks/useSendMessage.ts

```
@@ -64,8 +90,31 @@ function useSendMessage({ isRunning, setIsRunning, setExecutionStatus, message,
         options.signer = injector.signer;
         sourceAccount = account.address;
       }
-      await bridgeMessage.signAndSend(sourceAccount, { ...options });
-      setExecutionStatus(message.successfull);
+
+      const unsub = await bridgeMessage.signAndSend(sourceAccount, { ...options }, ({ events = [], status }) => {
+        if (status.isInBlock) {
+          events.forEach(({ event: { data, method } }) => {
+            if (method.toString() === 'MessageAccepted') {
+              const messageNonce = data.toArray()[1].toString();
+              sourceApi.rpc.chain.getBlock(status.asInBlock).then((res) => {
```

Error is not handled from this promise.